internal/pkg/cli/task_run.go

```diff
@@ -0,0 +1,242 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/config"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/docker/dockerfile"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type runTaskVars struct {
+	*GlobalOpts
+	Num    uint8
+	CPU    uint16
+	Memory uint16
+
+	Image          string
+	DockerfilePath string
+
+	TaskRole string
+
+	SubnetID         string
+	SecurityGroupIDs []string
+	App              string
+	Env              string
+
+	EnvVars  map[string]string
+	Commands string
+}
+
+type runTaskOpts struct {
+	runTaskVars
+
+	// Interfaces to interact with dependencies.
+	fs    afero.Fs
+	store store
+	df    dockerfileParser
+
+	// sets up Dockerfile parser using fs and input path
+	setupParser func(opts *runTaskOpts)
+}
+
+func newTaskRunOpts(vars runTaskVars) (*runTaskOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("new config store: %w", err)
+	}
+
+	return &runTaskOpts{
+		runTaskVars: vars,
+
+		fs:    &afero.Afero{Fs: afero.NewOsFs()},
+		store: store,
+
+		setupParser: func(o *runTaskOpts) {
+			o.df = dockerfile.New(o.fs, o.DockerfilePath)
+		},
+	}, nil
+}
+
+// Validate returns an error if the flag values passed by the user are invalid.
+func (o *runTaskOpts) Validate() error {
+	if o.Num == 0 {
+		return errors.New("number of tasks must be positive")
+	}
+
+	if o.CPU == 0 {
+		return errors.New("CPU units must be positive")
+	}
+
+	if o.Memory == 0 {
+		return errors.New("memory must be positive")
+	}
+
+	if o.Image != "" && o.DockerfilePath != "" {
+		return errors.New("cannot specify both image and dockerfile path")
+	}
+
+	if o.Image != "" {
+		if err := o.validateImageName(); err != nil {
+			return err
+		}
+	}
+
+	if o.DockerfilePath != "" {
+		if _, err := o.fs.Stat(o.DockerfilePath); err != nil {
+			return err
+		}
+	}
+
+	if o.Env != "" && (o.SubnetID != "" || o.SecurityGroupIDs != nil) {
+		return errors.New("can only specify one of a)env and b)subnet id and (or) security groups")
```

Comment:
Suggested rewording: "Either SubnetID or Security Group ID must be specified if environment is specified."

Reply:
The idea was that if environment is specified, then specifying either subnet-id or security-group-ids won't make sense, since environment is already associated with a VPC. 

I will change the wording to "neither SubnetID nor Security Group ID can be specified if environment is specified". It does sound so much better than the original one 😝